internal/controllers/synthetic_devices_controller.go

```diff
@@ -203,13 +221,9 @@ func (vc *SyntheticDevicesController) MintSyntheticDevice(c *fiber.Ctx) error {
 		return fiber.NewError(fiber.StatusBadRequest, "invalid vehicle id provided")
 	}
 
-	if _, err = vc.verifyUserAddressAndNFTExist(c.Context(), user, vid, rawIntegrationNode); err != nil {
-		return err
-	}
-
-	integration, err := vc.deviceDefSvc.GetIntegrationByTokenID(c.Context(), integrationNode)
+	vNFT, err := vc.verifyUserAddressAndNFTExist(c.Context(), user, vid, rawIntegrationNode)
```

Comment:
I think we should bail on this if `vNFT.UserDeviceID == nil`. It's unfortunate that this can happen but we don't have a burn function for vehicles yet.

Reply:
Can you handle this in the next PR? I'm going to merge this.